### PR TITLE
feat: refine cbc header

### PR DESCRIPTION
**Pull request by rafasdc**
Implements [NDT-386]

- [ ] Check to trigger automatic release process

[NDT-386]: https://connectivitydivision.atlassian.net/browse/NDT-386?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/components/Analyst/CBC/CbcEditProjectNumber.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    } catch (e) {
+      setError(
+        'An error occurred while attempting to validate the project number.'
+      );
+      return false;
+    }
+  };
+
+  const validateProjectNumber = async (val) => {
+    setIsValidating(true);
```
Comment (RRanath) [inline app/components/Analyst/CBC/CbcEditProjectNumber.tsx:154]: I think we need to validate if the project number is <4 as well?

> When editing and on read only display If “Project Number” is not 4 digits integer, validate with red box outline and text.  “Invalid number, please use a 4 digit number”.

File: db/revert/tables/cbc_add_fk_update_constraint.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-- Revert ccbc:tables/cbc_add_fk_update_constraint from pg
+
+BEGIN;
+
+-- XXX Add DDLs here.
```
Comment (RRanath) [inline db/revert/tables/cbc_add_fk_update_constraint.sql:5]: Do we need the revert script here?

File: app/components/Analyst/CBC/CbcEditProjectNumber.tsx [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  const [isSaving, setIsSaving] = useState(false);
+  const [updateProjectNumber] = useUpdateCbcProjectNumberMutation();
+
+  const [projectNumber, setProjectNumber] = useState(
+    queryFragment?.projectNumber
+  );
+
+  const [error, setError] = useState(null);
+
+  const handleSubmit = () => {
```
Comment (RRanath) [inline app/components/Analyst/CBC/CbcEditProjectNumber.tsx:85]: Noticed that after update the project number in dashboard it still shows the old project number.

File: app/components/Analyst/CBC/CbcEditProjectNumber.tsx (hunk 1 above)
Comment (rafasdc) [reply app/components/Analyst/CBC/CbcEditProjectNumber.tsx:154]: I thought so too at first, but AC states "Allow user to save the the project number with >=1 digit.", so current validation should be ok.

**Review by RRanath — APPROVED** [review]

**Pull request merged**